formatter error

## Setup

### Versions

- **Rust:** 1.32.0
- **Diesel:** 1.4.1
- **Database:** Postgres 11
- **Operating System** Linux

### Feature Flags

- **diesel:** chrono, postgres, r2d2, uuid

## Problem Description

On connecting to a DB and `embedded_migrations::run(&pool.get()?)?;`-ing, I get the incredibly unhelpful error "formatter error," which seems to be from [deep in std](https://doc.rust-lang.org/src/std/io/mod.rs.html#1146), if the migrations would introduce a constraint violation.

### What are you trying to accomplish?

Apply some migrations to prod that worked in dev

### What is the expected output?

[INFO] Started!

### What is the actual output?

[ERROR] Failed with: formatter error

### Are you seeing any additional errors?

Not from the app/diesel, but from diesel migration run, which shows the error.

### Steps to reproduce

have a DB
ALTER TABLE some check constraints on in a migration
run the migration

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)

--- a/diesel_migrations/migrations_internals/src/migration.rs
+++ b/diesel_migrations/migrations_internals/src/migration.rs
@@ -45,12 +45,12 @@ pub fn file_name<'a>(migration: &'a dyn Migration, sql_file: &'a str) -> Migrati
 
 impl<'a> fmt::Display for MigrationFileName<'a> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let fpath = match self.migration.file_path() {
-            None => return Err(fmt::Error),
-            Some(v) => v.join(self.sql_file),
-        };
-        f.write_str(fpath.to_str().unwrap_or("Invalid utf8 in filename"))?;
-        Ok(())
+        if let Some(path) = self.migration.file_path() {
+            let fpath = path.join(self.sql_file);
+            f.write_str(fpath.to_str().unwrap_or("Invalid utf8 in filename"))
+        } else {
+            write!(f, "{}/{}", self.migration.version(), self.sql_file)
+        }
     }
 }
 

